### PR TITLE
A fix for OneOf handling

### PR DESCRIPTION
**Pull request by Fitblip**
Depending on the schema version, type can be an array or a dictionary. This should handle both and allow for proper type comparisons with the new schema version. 

I also changed getattr to hasattr, since the latter won't throw a ValueError if it's attribute isn't found (which I believe was the intended check). 

Sorry about the soft quotes -> hard quotes too, I can update my PR if you want.

Maybe related to #9? I doubt it though, since they're using the old schema. /me shurgs.

### CODE REVIEW

**Comment by cwacek** [comment]
Thanks. It looks like this caused a couple test failures. If you can track those down I'll be able to merge it faster.

File: python_jsonschema_objects/classbuilder.py [hunk 1]
```diff
@@ -24,16 +24,18 @@ class ProtocolBase( collections.MutableMapping):
...
-          propval = getattr(self, prop)
+        out = {}
+        for prop in self:
+            propval = getattr(self, prop)
 
-          if isinstance(propval, list):
+            if isinstance(propval, list):
               out[prop] = [x.as_dict() for x in propval]
-          elif propval is not None:
+            elif not isinstance(propval, self):
```
Comment (cwacek) [inline python_jsonschema_objects/classbuilder.py]: methinks this should be `isinstance(propval, ProtocolBase)`, but I don't know what's causing the other test error.

**Comment by Fitblip** [comment]
Hey @cwacek, yeah there were a few issues with this. It was indeed supposed to be ProtocolBase, but I also realized that I wasn't using the most recent codebase (I was using what was on pip). I'll de-dupe and make sure tests pass, then squash it all into one commit - sound good?

**Comment by cwacek** [comment]
Works for me. Thanks!

**Comment by Fitblip** [comment]
Hey @cwacek - this should be fixed and passing all tests now. Cheers!

**Comment by Fitblip** [comment]
woot, thanks dude!

**Comment by cwacek** [comment]
No, thank you :)

**Comment by Fitblip** [comment]
Random question - does this automatically propagate to pypi? I want to make sure before I change my requirements.txt :)

**Comment by cwacek** [comment]
It does not. I'm in the process of updating it on pypi, but ran into a
thicket of problems on my machine. I should get it updated tonight.

Chris

**Comment by Fitblip** [comment]
Cool man, no worries! Just curious :). 

It actually looks like TravisCI can auto-cut new releases to Pypi if you want! http://docs.travis-ci.com/user/deployment/pypi/

Cheers!

**Comment by cwacek** [comment]
Good to know. For now I updated it manually.

**Pull request merged**